tmt/trying.py

```diff
@@ -0,0 +1,316 @@
+                return {
+                    "t": "test",
+                    "l": "login",
+                    "d": "discover",
+                    "p": "prepare",
+                    "e": "execute",
+                    "r": "report",
+                    "f": "finish",
+                    "k": "keep",
+                    "q": "quit",
+                    }[answer]
+            except KeyError:
+                self.info(click.style(f"Invalid action '{answer}'.", fg="red"))
+
+    def action_start(self, plan: Plan) -> None:
+        """ Common start actions """
+        plan.header()
+        plan.wake()
+
+    def action_start_test(self, plan: Plan) -> None:
+        """ Start with testing """
+        self.action_start(plan)
+
+        plan.discover.go()
+        plan.provision.go()
+        plan.prepare.go()
+        plan.execute.go()
+
+    def action_start_login(self, plan: Plan) -> None:
+        """ Start with login """
+        self.action_start(plan)
+
+        plan.provision.go()
+        plan.prepare.go()
+        assert plan.login is not None  # Narrow type
+        plan.login.go(force=True)
+
+    def action_start_ask(self, plan: Plan) -> None:
+        """ Ask what to do """
+        self.action_start(plan)
+
+        plan.provision.go()
+
+    def action_test(self, plan: Plan) -> None:
+        """ Test again """
+        plan.discover.go(force=True)
+        plan.execute.go(force=True)
+
+    def action_login(self, plan: Plan) -> None:
+        """ Log into the guest """
+        assert plan.login is not None  # Narrow type
+        plan.login.go(force=True)
+
+    def action_discover(self, plan: Plan) -> None:
+        """ Discover tests """
+        plan.discover.go(force=True)
+
+    def action_prepare(self, plan: Plan) -> None:
+        """ Prepare the guest """
+        plan.prepare.go(force=True)
+
+    def action_execute(self, plan: Plan) -> None:
+        """ Execute tests """
+        plan.execute.go(force=True)
+
+    def action_report(self, plan: Plan) -> None:
+        """ Report results """
+        plan.report.go(force=True)
+
+    def action_finish(self, plan: Plan) -> None:
+        """ Clean up guests and finish """
+        plan.finish.go()
+
+    def action_keep(self, plan: Plan) -> None:
+        """ Keep run and exit the session """
+        assert plan.my_run is not None  # Narrow type
+        run_id = click.style(plan.my_run.workdir, fg="magenta")
+        self.info(f"Run {run_id} kept unfinished. See you soon!")
+
+    def action_quit(self, plan: Plan) -> None:
+        """ Clean up the run and quit the session """
+
+        # Finish the run unless already done
+        if plan.finish.status() != "done":
+            plan.finish.go()
+
+        # Mention the run id and say good bye
+        assert plan.my_run is not None  # Narrow type
+        run_id = click.style(plan.my_run.workdir, fg="magenta")
+        self.info(f"\nRun {run_id} successfully finished. Bye for now!")
```

Comment:
I'm sorry, but the leading `\n` is truly unexpected. Plus, `action_keep` prints a similar message, without such leading `\n`. Can we drop it from here as well? :)

Reply:
Fixed in 842df674.